minor input and ui updates

Himanshu-01 opened the pull request:
* Add Splitscreen Mode into video settings to allow users to switch b/w horizontal & vertical modes
* Allow swapping grenades with 'A' in Bumper-Jumper layout

Himanshu-01 committed:
remove redundant controller structures and functions

Himanshu-01 committed:
allow switching grenades from `button A` in jumpy

Himanshu-01 committed:
Add `splitscreen_mode` setting ui option

Berthalamew commented on xlive/Blam/Engine/main/main_render.cpp:
```diff
@@ -79,7 +81,15 @@ void __cdecl main_render_player_view(void)
 		const bool use_horizontal_split = g_debug_render_horizontal_splitscreen || rasterizer_get_display_type() == _display_type_4_by_3;
-		display_split_type = use_horizontal_split ? _display_split_type_horizontal : _display_split_type_vertical;
+		
+		if(H2Config_split_mode == e_override_splitscreen_mode::automatic)
```
This code should go before the horizontal split, the global should override the Config setting if it's set

Himanshu-01 replied:
should just remove that horizontal boolean imo

Himanshu-01 committed:
Add split option menu to multiplayer video settings

 - use global strings to get default text instead of menu strings

Himanshu-01 committed:
Hide splitscreen setting in mp-video based on local player count

 - use existing e_display_split_type enum
 - force g_debug_render_horizontal_splitscreen

Berthalamew commented on xlive/Blam/Engine/interface/screens/screen_splitscreen_setting.cpp:
```diff
@@ -0,0 +1,287 @@
+		return _display_split_type_horizontal;
+		break;
+	}
+	return _display_split_type_none;
```
use a result variable instead of using a return for every case.

Berthalamew commented on xlive/Blam/Engine/interface/screens/screen_splitscreen_setting.h:
```diff
@@ -0,0 +1,60 @@
+
+/* constants */
+/* enums */
+enum
```
This should go under constants

Berthalamew commented on xlive/Blam/Engine/interface/screens/screen_video_settings.h:
```diff
@@ -13,7 +13,7 @@
 enum
 {
-	k_no_of_visible_items_for_video_settings = (8 + 1)
+	k_no_of_visible_items_for_video_settings = (8 + 2)
```
replace this with `k_no_of_visible_items_for_video_settings = k_total_no_of_video_settings_list_items` ?

Himanshu-01 replied:
nah , it might be same for this context but they both symbolize different things, better to keep them seperate

Berthalamew replied:
alright

Berthalamew commented on xlive/H2MOD/Modules/Shell/Config.cpp:
```diff
@@ -8,6 +8,7 @@
 #include "render/render_lod_new.h"
 #include "rasterizer/dx9/rasterizer_dx9.h"
 #include "rasterizer/dx9/rasterizer_dx9_9on12.h"
+#include <render/render.h>
```
#include <render/render.h> -> #include "render/render.h"

Berthalamew commented on xlive/H2MOD/Modules/Shell/Config.cpp:
```diff
@@ -494,6 +496,7 @@ void SaveH2Config()
 			CONFIG_SET(&ini, "use_d3d9on12", &g_rasterizer_dx9on12_enabled);
 			CONFIG_SET(&ini, "disable_amd_or_ati_patches", &g_rasterizer_dx9_driver_globals.disable_amd_or_ati_patches);
 			CONFIG_SET(&ini, "use_vsync", &H2Config_use_vsync);
+			CONFIG_SET(&ini, "splitscreen_mode", std::to_string(H2Config_split_mode).c_str());
```
why is this being converted to a string?
Shouldn't we just cast this to an integer and then use the set functionality for that?

Himanshu-01 committed:
add requested fixes

Berthalamew committed:
make k_splitscreen_options_string static

Berthalamew committed:
remove tag_loaded call and use the index in the shared globals
